solr/solrj/src/java/org/apache/solr/common/cloud/SolrZkClient.java

```diff
@@ -241,7 +241,7 @@ public void delete(final String path, final int version, boolean retryOnConnLoss
       throws InterruptedException, KeeperException {
     if (retryOnConnLoss) {
       zkCmdExecutor.retryOperation(() -> {
-        keeper.delete(path, version);
+        getSolrZooKeeper().delete(path, version);
```

Comment:
switched all retry operations to resolve `keeper` at call time (so a retry will re-resolve the object)